File: java/src/main/java/io/yeeco/yeesigner/ErrorUtils.java
package io.yeeco.yeesigner;

import java.util.HashMap;
import java.util.Map;

public class ErrorUtils {

    private static Map<Byte, String> errorMessage = new HashMap<Byte, String>() {{
        put((byte) 1, "unknown");
        put((byte) 2, "invalid mini secret key");
        put((byte) 3, "invalid secret key");
        put((byte) 4, "invalid public key");
        put((byte) 5, "invalid signature");
        put((byte) 6, "jni error");
        put((byte) 7, "invalid method");
        put((byte) 8, "invalid tx");
        put((byte) 9, "invalid json");
    }};

    public static void checkErrorCode(byte code) throws SignerException {

        if (code != 0) {
            String message = errorMessage.get(code);
            message = message != null ? message : "unknown";
            throw new SignerException(message);

        }

    }

}

File: oc/YeeSigner/YeeSigner/YeeSigner/ErrorUtils.m
//
//  ErrorUtils.m
//  YeeSigner
//
//  Created by GB on 2020/6/20.
//  Copyright © 2020 io.yeeco. All rights reserved.
//

#import "ErrorUtils.h"

@implementation ErrorUtils

+ (NSError* ) error:(unsigned int )err {
    NSDictionary* dict = @{
        @1: @"unknown",
        @2: @"invalid mini secret key",
        @3: @"invalid secret key",
        @4: @"invalid public key",
        @5: @"invalid signature",
        @6: @"jni error",
        @7: @"invalid method",
        @8: @"invalid tx",
        @9: @"invalid json",
        @10: @"address encode error",
        @11: @"address decode error",
    };
    
    NSString *message = [dict objectForKey:@(err)];
    message = message ==nil ? @"unknown" : message;
    
    return [NSError errorWithDomain:@"YeeSigner" code:err userInfo:@{@"message": message}];
}

@end
